src/runner.js

```diff
@@ -41,74 +55,180 @@ TODO?  Promises or callback as first arg?  Are we node?
 
 let actions = require("./common/actions");
 let chai = require("chai");
+let type = require("./jetpack/type");
+
+const config = {
+  alwaysRun: false
+};
 
-let config = exports.config = {};
+
+let validate = function (thing, rules) {
+  // TODO, this is way wack.  Reinvent wheels much?  return order?
+  let errors  = [];
+  rules.map(function(rule) {
+    try {
+      rule(thing);
+    } catch (E) {
+      errors.push(E);
+    }
+  });
+  return ([errors, errors.length === 0]);
+};
 
 // TODO, write this, decide of return val (throw? false?  list of errors?)
 // TODO, use an existing validation system?  chai maybe?
 // TODO, record and describe failures
 let validateConfig = function (config) {
   // I think this should return an errors, status object, not a bool.
-  let errors  = [];
   let rules = [
     function (c) {chai.expect(c.name).to.be.a("string")},
     function (c) {chai.expect(c.description).to.be.a("string")},
-    function (c) {chai.expect(c.recipe).to.be.a("function")},
+    function (c) {chai.expect(c.owner).to.be.a("string")},
+    function (c) {chai.expect(c.version).to.be.a("number")},
+    function (c) {chai.expect(c.run).to.be.a("function")},
     function (c) {chai.expect(c.shouldRun).to.be.a("function")}
   ];
```

Comment:
should we add / require a 'guarantee'  or 'rollback', per @lauraxt.  
implied:
- `bool guarantee(state)` <- the state changed?  this is the claim that the recipe makes.  See #33